dapr/proto/components/v1/state.proto

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+syntax = "proto3";
+
+package dapr.proto.components.v1;
+
+import "google/protobuf/any.proto";
+import "dapr/proto/common/v1/common.proto";
+import "dapr/proto/components/v1/common.proto";
+
+option go_package = "github.com/dapr/dapr/pkg/proto/components/v1;components";
+
+
+// QueriableStateStore service provides a gRPC interface for querier state store components.
+service QueriableStateStore {
+  // Query performs a query request on the statestore.
+  rpc Query(QueryRequest) returns (QueryResponse) {}
+}
+
+message Sorting {
+  // The key that should be used for sorting.
+  string key = 1;
+  enum Order {
+    ASC = 0;
+    DESC = 1;
+  }
+  // The order that should be used.
+  Order order = 2;
+}
+
+message Pagination {
+  // Maximum of results that should be returned.
+  int64 limit = 1;
+  // The pagination token.
+  string token = 2;
+}
+
+message Query {
+  // Filters that should be applied.
+  map<string, google.protobuf.Any> filter = 1;
+  // The sort order.
+  repeated Sorting sort = 2;
+  // The query pagination params.
+  Pagination pagination = 3;
+}
+
+// QueryRequest is for querying state store.
+message QueryRequest {
+  // The query to be performed.
+  Query query = 1;
+  // Request associated metadata.
+  map<string, string> metadata = 2;
+}
+
+// QueryItem is an object representing a single entry in query results.
+message QueryItem {
+  // The returned item Key.
+  string key = 1;
+  // The returned item Data.
+  bytes data = 2;
+  // The returned item ETag
+  dapr.proto.common.v1.Etag etag = 3;
+  // The returned error string.
+  string error = 4;
+  // The returned contenttype
+  string content_type = 5;
+}
+
+// QueryResponse is the query response.
+message QueryResponse {
+  // The query response items.
+  repeated QueryItem items = 1;
+  // The response token.
+  string token = 2;
+  // Response associated metadata.
+  map<string, string> metadata = 3;
+}
+
+// TransactionalStateStore service provides a gRPC interface for transactional state store components.
+service TransactionalStateStore {
+    // Multi executes multiples operation in a transactional environment.
+    rpc Multi(TransactionalStateRequest) returns (TransactionalStateResponse) {}
```

Comment:
Why are we calling this operation/method `Multi`?

Reply:
Good point, I just copy/paste the same as used in code [here](https://github.com/dapr/components-contrib/blob/master/state/transactional_store.go#L19). 

I have no objection to changing it to something more suitable ideas?, "rpc ExecTransaction" "rpc Transaction" "rpc Transact"

Reply:
I agree. Can we change to Transact?

Reply:
Fixed